### DISCUSSION TITLE
OpenTelemetry user location not working in Application Insights

### DISCUSSION THREAD

**Discussion by raix** [microsoft/aspire#1645] [Q&A]
When deploying behind a proxy YARP or Container Apps / Envoy we need to use forwarded headers.

On PlatformPlatform we have a [POC pull-request for enriching OpenTelemetry data with client ip](https://github.com/platformplatform/PlatformPlatform/blob/167e78515ef5aeb91446ed85ce9e3df28f931ad3/application/shared-kernel/ApiCore/Aspire/ServiceDefaultsExtensions.cs#L70) _(will be discarded as it's not supported)_

The client ip turns up as a custom property in Application Insights - showing the actual user ip address and not as resolved location + anonymized ip "0.0.0.0".

It seems like OpenTelemetry support in Azure Application Insights are not ready for production yet / early days for OT support?

Any ETA or timeline on when support in Azure is expected?

Kind regards

Morten

**Comment by DamianEdwards** [comment]
Seems like a question for the Application Insights/Azure Monitor folks. @samsp-msft can you help direct?
